src/vs/workbench/contrib/terminal/node/terminalProcess.ts

```diff
@@ -70,12 +70,22 @@ export class TerminalProcess implements ITerminalChildProcess, IDisposable {
 		};
 
 		try {
-			this._ptyProcess = pty.spawn(shellLaunchConfig.executable!, shellLaunchConfig.args || [], options);
-			this._processStartupComplete = new Promise<void>(c => {
-				this.onProcessIdReady((pid) => {
-					c();
+			const filePath = path.basename(shellLaunchConfig.executable!);
+			if (fs.existsSync(filePath)) {
+				this._ptyProcess = pty.spawn(shellLaunchConfig.executable!, shellLaunchConfig.args || [], options);
+				this._processStartupComplete = new Promise<void>(c => {
+					this.onProcessIdReady((pid) => {
+						c();
+					});
 				});
-			});
+			}
+			else {
+				// file path does not exist , handle it with negative exit code
+				this._exitCode = -1;
```

Comment:
Let's create a new constant like `SHELL_PATH_INVALID_EXIT_CODE` inside `common/terminal.ts` and compare against that on both sides.

Reply:
Done